apps/app/src/hooks/useUser.ts

```diff
@@ -0,0 +1,41 @@
+import { useMutation, useQuery, useQueryClient } from "@tanstack/react-query";
+import { getLoggedInUser, logoutUser } from "../services/users.service";
+import { UserData } from "../types";
+
+const USER_QUERY_KEY = ["user"];
+
+export const useUser = () => {
+	const queryClient = useQueryClient();
+	const { data: user, ...rest } = useQuery({
+		queryKey: USER_QUERY_KEY,
+		queryFn: async () => {
+			try {
+				const { data } = await getLoggedInUser({});
+
+				return data["data"]["_user"];
+			} catch (err) {
+				return null;
+			}
+		},
+		staleTime: Infinity,
+		retry: false,
+	});
+
+	const setUser = (user: UserData | null) => {
+		queryClient.setQueryData(USER_QUERY_KEY, user);
+	};
+
+	const updateLoggedInUser = useMutation(getLoggedInUser, {
```

Comment:
Czy ta funkcja jest potrzebna? `getLoggedInUser` to przecież NIE jest mutacja.
Jeśli chcesz pobrać na nowo użytkownika, to powinieneś zinwalidować cache dla `USER_QUERY_KEY` co spowoduje automatyczne pobranie danych na nowo.